offer-manager-domain/src/main/groovy/life/qbic/business/offers/policies/taxes/InternalTaxPolicy.java

```diff
@@ -3,23 +3,20 @@
 import java.math.BigDecimal;
 import life.qbic.business.persons.affiliation.AffiliationCategory;
 
-public class InternalTaxPolicy extends TaxPolicy {
+public class InternalTaxPolicy implements TaxPolicy {
 
-  public static InternalTaxPolicy of(AffiliationCategory category, String country) {
+  protected static InternalTaxPolicy of(AffiliationCategory category, String country) {
     if (category != AffiliationCategory.INTERNAL) {
       throw new PolicyViolationException("Cannot apply internal tax policy to affiliation of type " + category);
     }
     if (!country.equalsIgnoreCase("Germany")) {
       throw new PolicyViolationException(
           "Cannot apply internal tax policy to countries outside of Germany: " + country);
     }
-    return new InternalTaxPolicy(category, country);
+    return new InternalTaxPolicy();
   }
 
-  protected InternalTaxPolicy(
-      AffiliationCategory targetAffiliationCategory,
-      String country) {
-    super(targetAffiliationCategory, country);
+  private InternalTaxPolicy() {
   }
 
   @Override
```

Comment:
if the taxes are zero, the calculatedTaxes must also be zero, right?

Reply:
you mean if the tax rate is 0%, the calculated taxes from a price value are zero? That is a trivia, isnt it? :)

Reply:
Yes it is but needs to be applied :) I guess thats why the value in the offer is not correct currently

Reply:
I think its the same as for OutsideGermanyTaxPolicy. You need to return zero here

Reply:
ah,  i think it is contained in my refactor i pushed